zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RestErrorMapper.java

```diff
@@ -133,4 +133,16 @@ public static <T> ResponseEntity<T> mapProblemToResponse(final ProblemDetail pro
         .headers(httpHeaders -> httpHeaders.setContentType(MediaType.APPLICATION_PROBLEM_JSON))
         .build();
   }
+
+  public static ResponseEntity<Object> mapUserManagementExceptionsToResponse(final Exception e) {
+    if (e instanceof IllegalArgumentException) {
+      final ProblemDetail problemDetail =
+          ProblemDetail.forStatusAndDetail(HttpStatus.BAD_REQUEST, e.getMessage());
+      return ResponseEntity.of(problemDetail).build();
+    }
+
+    final ProblemDetail problemDetail =
+        ProblemDetail.forStatusAndDetail(HttpStatus.INTERNAL_SERVER_ERROR, e.getMessage());
+    return ResponseEntity.of(problemDetail).build();
+  }
```

Comment:
🔧 You can use the `createProblemDetail` and `mapProblemToResponse` methods for brevity and consistency (as this also forces you to add an appropriate `title` to the problem detail.